src/jdk.compiler/share/classes/com/sun/tools/javac/comp/TypeEnter.java

```diff
@@ -1337,9 +1337,17 @@ public Type constructorType() {
 
         @Override
         public MethodSymbol constructorSymbol() {
+            // we should do this only once
+            boolean copyConstAnnos = constructorSymbol == null;
```

Comment:
Question: can we avoid repeating this method completely if `constructorSymbol != null`? Seems a bit weird to not copy the annotation again (which I understand), but redo all the rest again.

Reply:
yep good point